ports/rp2/machine_rtc.c

```diff
@@ -0,0 +1,457 @@
+/*
+ * astrorafael port of RP2 Real Time Clock module, based on previous
+ * work by the MicroPython project
+ *
+ * This file is part of the MicroPython project, http://micropython.org/
+ *
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2020-2021 Damien P. George
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+// --------------------
+// MicroPython includes
+// --------------------
+
+#include "py/runtime.h"
+#include "py/mperrno.h"
+#include "py/mphal.h"
+#include "lib/utils/mpirq.h"
+
+// ----------------------
+// Pico specific includes
+// ----------------------
+
+#include "hardware/rtc.h"
+#include "pico/util/datetime.h"
+
+// ------------------
+// Exported constants
+// ------------------
+
+#define RTC_ALARM0 (0)
+
+// ------------------
+// Internal constants
+// ------------------
+
+/* 
+ * This should be defined and exported in the Python machine module
+ * but apparently it is not used
+ */
+#define RP2_PWR_MODE_IDLE         (0x01)
+#define RP2_PWR_MODE_SLEEP        (0x02)
+#define RP2_PWR_MODE_DEEPSLEEP    (0x04)
+
+enum dotw {Sunday = 0, Monday, Tuesday, Wednesday, Thursday, Friday, Saturday};
```

Comment:
Can you spell this out for readability?

Reply:
Do you mean writing a comment for dotw (day of the week) ?

Reply:
Rather just writing day_of_the_week or something like that: immediately clear and no comment needed

Reply:
ok